README
Dist::Zilla::Plugin::AutoVersion::Relative



INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Kent Fredric.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Plugin-AutoVersion-Relative

1.001002 2017-03-09T08:46:40Z 106a5b6

1.001001 2017-03-09T08:45:32Z f6ffa8b
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot
 - Remove use of ConfigDumper util
 - Remove use of Test::KENTNL/dztest

 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +13 ↑4 -3 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -2 (recommends: +1)
 - test: +1 ↓1 -2 (recommends: +1 ↑1 ↓1)

1.001000 2014-08-29T15:43:44Z
 [00 Minor]
 - Tests refactored
 - Minor internals changes.

 [Dependencies::Stats]
 - Dependencies changed since 1.000001, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - runtime: +2 -1
 - test: +2

 [Internals]
 - dump_config now uses Util::ConfigDumper
 - use Const::Fast instead of Readonly

 [Tests]
 - Tests refactored into using DZUTKENTNL::dztest

1.000001 2014-08-15T21:15:21Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - tests augmented.
 - metadata updates.
 - whitespace adjustment.

 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)
 - test: -3

 [Metadata]
 - 1x contributor added.

 [Misc]
 - POD rearranged source side to produce less whitespace at the top of generated code.

 [Tests]
 - Update Test::ReportPrereqs

1.000000 2014-07-31T05:37:28Z
 [00 Major]
 - Changes that may break downstreams who work by hand.

 [01 Version Scheme]
 - Uses x.yyyyyy (Mantissa 6) instead of x.yyyyyyyy ( Mantissa 8 )
 - This is the primary reason for the 1.x on the box.
 - https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [02 Toolkit]
 - Now uses ExtUtils::MakeMaker
 - "Latest" dependenceis are now only recommendations (softened)
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.03000002, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +44 -2 (recommends: -1, suggests: +1 ↑1)
 - runtime: +1 ↑1
 - test: +5 ↑1 ↓1 -2 (recommends: +3)

0.03000002 2013-04-09T02:50:19Z
 - Maintenance Release for MB 0.4004

 [Dependencies::Stats]
 - Dependencies changed since 0.03000001, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +5 (recommends: ↑1, suggests: ↑1)
 - test: ↑1

 [Documentation]
 - Update Copyright year
 - Add README.mkdn

 [Meta]
 - bugtracker to github issues

 [Packaging]
 - Utilise test_requires in Build.PL

0.03000001 2012-02-03T02:19:44Z
 [Bugs]
 - Added prototypes on all tokens so "foo + 1" is "foo() + 1" not "foo(+1)" ( Thanks Karen Etheridge )

0.03000000 2012-02-02T10:42:53Z
 [Features]
 - fixed "days" variable to accurately indicate the exact number of days past since the milestone (Karen Etheridge).
   For the old way of computing date, please use days_square instead ( available since 0.02000000 )

0.02000000 2012-02-02T10:23:49Z
 [Dependencies::Stats]
 - Dependencies changed since 0.01060309, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: +3
 - test: -1

 [Features]
 - Added "days_accurate" and "days_square" alternative to "days" field. "days_square" is the same as the previous
   "days" field, but now available separately for backwards compatibility.  Based on code from Karen Etheridge

 [Internals]
 - All namespaces declare $AUTHORITY="cpan:KENTNL";
 - Version declarations moved out of BEGIN

 [Packaging]
 - Toolkit Upgrade.
 - Update LICENSE ( Year, Indentation, Address )
 - Extra Tests moved to xt/
 - Repo urls all moved to https://
 - Declares x_authority
 - Uses git based versioning now.
 - Nolonger self-feeds via Bootstrap::lib

0.01060309 2011-04-05T21:49:08Z
 - Maintainence release only.

 [Dependencies::Stats]
 - Dependencies changed since 0.01037118, see misc/*.deps* for details
 - develop: +1 (recommends: +1, suggests: +1)
 - runtime: +2

 [Packaging]
 - Ship .perltidyrc
 - Rework Changes for CPAN::Changes form.
 - Minor crtiic cleanup.

 [Tests]
 - Added CPAN::Changes tests.
 - Removed portability tests.

0.01037118 2010-08-22T06:25:19Z
 - Fix namespace::autoclean deps up to get rid of CMOP fail tests.
 - Get rid of =inc::lib
 - Get rid of old critic test and autogen now
 - Add data to MetaConfig reporting.

 [Dependencies::Stats]
 - Dependencies changed since 0.01035617, see misc/*.deps* for details
 - runtime: ↑1
 - test: +1

0.01035617 2010-08-07T05:42:35Z
 - Depend on latest moose and Mx::StrictConstructor to resolve Deprecation problems.

 [Dependencies::Stats]
 - Dependencies changed since 0.01034313, see misc/*.deps* for details
 - runtime: ↑2 -1

0.01034313 2010-07-25T01:17:26Z
 - Reduced a few dependencies, cleaned surplus T-Deps.
 - Modified docs a little bit.

 [Dependencies::Stats]
 - Dependencies changed since 0.01027903, see misc/*.deps* for details
 - build: -7
 - runtime: -1
 - test: +6

0.01027903 2010-05-22T15:24:03Z
 - Moved to @KENTNL, + inc::lib bootstrap
 - Made compatible with DZIL 2.0+

 [Dependencies::Stats]
 - Dependencies changed since 0.01010513, see misc/*.deps* for details
 - build: +7 ↑1
 - configure: ↑1
 - runtime: ↑1

0.01010513 2009-12-04T01:46:57Z
 - Drop lots of downstream deps by using only ::DateTime
 - Reduced support for Olsons ( they're not that useful anyway, they change with daylight saving )
 - Fixed inc:: bootstrapping in build

 [Dependencies::Stats]
 - Dependencies changed since 0.01010014, see misc/*.deps* for details
 - runtime: +1 -1

0.01010014 2009-11-30T02:44:26Z
 - Fix Dep: Readonly

 [Dependencies::Stats]
 - Dependencies changed since 0.01010003, see misc/*.deps* for details
 - runtime: +1

0.01010003 2009-11-29T15:50:13Z
 - Reworked the documentation to utilize Pod::Weaver.
 - +perltidy
 - +perlcritic
 - +compiletests
 - -use ok
 - +waverconfig
 - Injected gitty goodness into dist.ini
 - Ported to Pod::Weaver
 - Drop MakeMaker support

 [Dependencies::Stats]

Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Time-Relative versioning",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-AutoVersion-Relative",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::AutoVersion::Relative",
  "PREREQ_PM" => {
    "Const::Fast" => 0,
    "DateTime" => "0.50",
    "Dist::Zilla" => "2.10131",
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "Moose" => "1.09",
    "MooseX::StrictConstructor" => "0.10",
    "MooseX::Types::DateTime" => "0.05",
    "MooseX::Types::Moose" => 0,
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla" => "2.10131",
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.001002",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Const::Fast" => 0,
  "DateTime" => "0.50",
  "Dist::Zilla" => "2.10131",
  "Dist::Zilla::Plugin::GatherDir" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "Dist::Zilla::Role::VersionProvider" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Moose" => "1.09",
  "MooseX::StrictConstructor" => "0.10",
  "MooseX::Types::DateTime" => "0.05",
  "MooseX::Types::Moose" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "namespace::autoclean" => "0.09",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
  metadata KiokuDB versioning plugin tuneables cldr CLDR cultin cultin
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');

#$bundle->remove_policy('CodeLayout::RequireUseUTF8');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4206"
  },
  "configure_requires" => {
    "Module::Build" => "0.4206"
  },
  "dist_abstract" => "Time-Relative versioning",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-AutoVersion-Relative",
  "dist_version" => "1.000000",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::AutoVersion::Relative",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "DateTime" => "0.50",
    "Dist::Zilla" => "2.10131",
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "Moose" => "1.09",
    "MooseX::StrictConstructor" => "0.10",
    "MooseX::Types::DateTime" => "0.05",
    "MooseX::Types::Moose" => 0,
    "Readonly" => 0,
    "namespace::autoclean" => "0.09",
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Dist::Zilla" => "2.10131",
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::Tester" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Scalar::Util" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "Dist::Zilla" => "2.10131",
  "Dist::Zilla::Plugin::GatherDir" => 0,
  "Dist::Zilla::Tester" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4206",
  "Scalar::Util" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
